python/pyspark/rdd.py

```diff
@@ -2067,7 +2067,7 @@ def add_shuffle_key(split, iterator):
                     avg = int(size / n) >> 20
                     # let 1M < avg < 10M
                     if avg < 1:
-                        batch *= 1.5
```

Comment:
Actually when `get_used_memory() > limit` is true, I don't know why we want to increase `batch *= 1.5`.

Reply:
I guess to increase the size of batch and to use more memory .. ?

Reply:
Hm..I thought increasing `batch` is for `c > batch`. In other words, it increases the size of batch if it reaches the current batch size, but used memory is still under `limit` (and the average size of bucket is small).

If it reaches memory limit before reaching the batch size, it seems not make sense to increase batch size (even the average size of bucket is small).

Reply:
Agree. the batch size should not increase when reaching the memory limit